storage/Filesystems/MountableImpl.cc

```diff
@@ -242,19 +250,27 @@ namespace storage
     {
 	const Storage& storage = actiongraph.get_storage();
 
-	EtcFstab fstab(storage.get_impl().prepend_rootprefix("/etc"));	// TODO pass as parameter
+	EtcFstab fstab;
+        fstab.read(storage.get_impl().prepend_rootprefix("/etc"));	// TODO pass as parameter
+
+        FstabEntry * entry = new FstabEntry();
+        entry->set_device(get_mount_by_name());
+        entry->set_mount_point(mountpoint);
+        entry->set_mount_opts(get_mount_opts());
+
+        // FIXME: Why is this not a FsType in the first place?
```

Comment:
Likely because the old EtcFstab class expected a string. I would propose to move the FsType enum from Filesystem to Mountable.